qiita_db/sql_connection.py

```diff
@@ -1047,10 +1067,54 @@ def rollback(self):
         except Exception:
             self._connection.close()
             raise
+        # Execute the post rollback functions
+        self._funcs_executor(self._post_rollback_funcs, "rollback")
 
     @property
     def index(self):
         return len(self._queries) + len(self._results)
 
+    @_checker
+    def add_post_commit_func(self, func):
+        """Adds a post commit function
+
+        The function added will be executed after the next commit in the
+        transaction, unless a rollback is executed. This is useful, for
+        example, to perform some filesystem clean up once the transaction is
+        committed.
+
+        Parameters
+        ----------
+        func : function
+            The function to add for the post commit functions
+
+        Notes
+        -----
+        func should not accept any parameter, i.e. it should allow to be
```

Comment:
This does not concern that specific function. I could see this functionality being useful if, inside a transaction, we are moving files, and if the transaction fails, the file list is passed on rollback to move them all back to where they need to be. This way we have both filesystem and database integrity, and the ability to rollback files by tracking that list. So maybe not as useful for post commit, but definitely rollback.